internal/component/prometheus/remote/queue/serialization/serializer_bench_test.go

```diff
@@ -0,0 +1,112 @@
+package serialization
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/alloy/internal/component/prometheus/remote/queue/types"
+	"github.com/prometheus/prometheus/model/labels"
+)
+
+var lbls = labels.FromStrings("one", "two", "three", "four")
+
+func BenchmarkAppender(b *testing.B) {
+	// This should be 0 allocs
+	b.ReportAllocs()
+	logger := log.NewNopLogger()
+	for i := 0; i < b.N; i++ {
+		app := NewAppender(context.Background(), 1*time.Hour, &fakeSerializer{}, logger)
+		for j := 0; j < 10_000; j++ {
+			_, _ = app.Append(0, lbls, time.Now().Unix(), 1.1)
+		}
+		_ = app.Commit()
+	}
+}
+
+func BenchmarkSerializer(b *testing.B) {
+	// This should be around 200-300 allocs 7m ns/op
+	series := getTimeSeries(b)
+	b.ResetTimer()
+	b.ReportAllocs()
+	logger := log.NewNopLogger()
+	for i := 0; i < b.N; i++ {
+		serial, _ := NewSerializer(types.SerializerConfig{
+			MaxSignalsInBatch: 1_000,
+			FlushFrequency:    1 * time.Second,
+		}, &fakeFileQueue{}, func(stats types.SerializerStats) {
```

Comment:
Hmmm... don't think we need to benchmark NewSerializer.
Also, shouldn't we use `b.N` to determine number of series that will be passed to `SendSeries`?

Reply:
I was worried about the overhead of creating a series, but seems minimal. Mostly concerned with allocs since that tends to be the cpu killer for any implementation of the wal. Changed to be more accurate of a benchmark, good call.